code/game/machinery/camera/camera.dm

```diff
@@ -46,20 +46,23 @@ var/list/camera_names=list()
 		ASSERT(src.network.len > 0)
 
 	if(!c_tag)
-		var/area/A=get_area(src)
-		var/basename=A.name
-		var/nethash=english_list(network)
-		var/suffix = 0
-		while(1)
-			c_tag = "[basename]"
-			if(suffix)
-				c_tag += " [suffix]"
-			if(!(nethash+c_tag in camera_names))
-				camera_names[nethash+c_tag]=src
-				break
-			suffix++
+		name_camera()
 	..()
 
+/obj/machinery/camera/proc/name_camera()
+	var/area/A=get_area(src)
+	var/basename=A.name
+	var/nethash=english_list(network)
+	var/suffix = 0
+	while(1)

```

Comment:
This could just be while(nethand+c_tag in camera_names). It has the same effect as the break, but saner.